src/test/app/Vault_test.cpp

```diff
@@ -2859,6 +2859,44 @@ class Vault_test : public beast::unit_test::suite
             env(tx1);
         });
 
+        testCase([&, this](
+                     Env& env,
+                     Account const& owner,
+                     Account const& issuer,
+                     Account const& charlie,
+                     auto,
+                     Vault& vault,
+                     PrettyAsset const& asset,
+                     auto&&...) {
+            testcase("IOU calculation rounding");
+
+            auto [tx, keylet] = vault.create({.owner = owner, .asset = asset});
+            tx[sfScale] = 1;
+            env(tx);
+            env.close();
+
+            env(vault.deposit(
+                {.depositor = owner, .id = keylet.key, .amount = asset(100)}));
+            env.close();
+
+            auto const tx1 = vault.deposit(
+                {.depositor = owner,
+                 .id = keylet.key,
+                 .amount = asset(Number(375, -2))});
+            for (auto i = 0; i < 5; ++i)
+            {
+                env(tx1);
+            }
+            env.close();
+
+            env(vault.withdraw(
+                {.depositor = owner,
+                 .id = keylet.key,
+                 .amount = asset(Number(1000 + 37 * 5, -1))}));
```

Comment:
This could use a brief explanation of what you're intending to check.

It looks like you're making 5 deposits of 3.75 each, which should be a total of 18.75. But the vault has a scale of 1, so each one gets rounded to 3.7, which is a total of 18.5? A `BEAST_EXPECT` or two verifying the vault balance, and/or the `owner` balance would be good in between the deposits and withdraw.

Then you're withdrawing 118.5, which is way more than was deposited. So are you intentionally overpaying, so that it gets truncated? Or is the 1000 a mistake? Either way, you've withdrawn the entire balance, which is shown(?) by the fact that you can delete the vault on the next line.

Reply:
added comments in d062bb6d24